packages/core/src/apiml/Services.ts

```diff
@@ -207,7 +207,26 @@ export class Services {
 
         const _genCommentsHelper = (key: string, elements: string[]): string => {
             if (elements == null || elements.length === 0) return "";
-            return `//"${key}": "${elements.length === 1 ? elements[0] : elements.join('"\n//"' + key + '": "')}"`;
+            let kvPair: string = "";
+            let endComma = "";
+            // add comma to end of kvp depending on if base profile
+            if(!key.includes("base")){
+                endComma = ",";
+            }
+            if (elements.length === 1){
+                return `//"${key}": "${elements[0]}"${endComma}`;
+            }
+            // format pair of kvp (add commas in expected places)
+            elements.forEach((element, index) => {
+                if (index === elements.length - 1) {
+                    kvPair += `\n//"${key}": "${element}"`;
+                } else {
+                    kvPair += `\n//"${key}": "${element}"${endComma}`;
+                }
+            });
```

Comment:
Not required, but `kvPair` can be assigned to the result of `elements.reduce`, which would allow `kvPair` to be reassigned as a constant variable:

```suggestion
            const kvPair = elements.reduce((all, current: string, index) => {
                return all.concat(index === elements.length - 1 ? `\n//"${key}": "${element}"` :  `\n//"${key}": "${element}"${endComma}`);
            }, "");
```